miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -188,16 +200,33 @@ internal class MiniAppDownloader(
     @SuppressWarnings("LongMethod")
     private suspend fun downloadMiniApp(
         miniAppInfo: MiniAppInfo,
-        manifest: ManifestEntity
+        manifest: Pair<ManifestEntity, ManifestHeader>
     ): String {
         val appId = miniAppInfo.id
         val versionId = miniAppInfo.version.versionId
         val baseSavePath = storage.getMiniAppVersionPath(appId, versionId)
         when {
-            isManifestValid(manifest) -> {
-                for (file in manifest.files) {
-                    val response = apiClient.downloadFile(file)
-                    storage.saveFile(file, baseSavePath, response.byteStream())
+            isManifestFileExist(manifest.first) -> {
+                for (file in manifest.first.files) {
+                    if (isRequireSignatureVerification) {
+                        if (isSignatureValid(apiClient.downloadFile(file).byteStream(), versionId, manifest)) {
+                            storage.saveFile(file, baseSavePath, apiClient.downloadFile(file).byteStream())
+                            miniAppAnalytics.sendAnalytics(
+                                    eType = Etype.CLICK,
+                                    actype = Actype.SIGNATURE_VALIDATION_SUCCESS,
+                                    miniAppInfo = miniAppInfo
+                            )
+                        } else {
+                            removeMiniApp(miniAppInfo, "$SIGNATURE_VERIFICATION_ERR " +
+                                    "The files will be deleted.")
+                            miniAppAnalytics.sendAnalytics(
+                                    eType = Etype.CLICK,
+                                    actype = Actype.SIGNATURE_VALIDATION_FAIL,
+                                    miniAppInfo = miniAppInfo
+                            )
+                            throw MiniAppVerificationException(SIGNATURE_VERIFICATION_ERR)
+                        }
+                    } else storage.saveFile(file, baseSavePath, apiClient.downloadFile(file).byteStream())
```

Comment:
Even when `isRequireSignatureVerification` is set to false, we still want to check it with the Verifier. That's why we want to send analytics event: so we can track which devices the verification fails on before enabling verification to be required for all mini apps. For example:

```suggestion
                    if (isSignatureValid(apiClient.downloadFile(file).byteStream(), versionId, manifest)) {
                        miniAppAnalytics.sendAnalytics(
                                eType = Etype.CLICK,
                                actype = Actype.SIGNATURE_VALIDATION_SUCCESS,
                                miniAppInfo = miniAppInfo
                        )
                    } else {
                        miniAppAnalytics.sendAnalytics(
                                eType = Etype.CLICK,
                                actype = Actype.SIGNATURE_VALIDATION_FAIL,
                                miniAppInfo = miniAppInfo
                        )
                    
                        if (isRequireSignatureVerification) {
                            removeMiniApp(miniAppInfo, "$SIGNATURE_VERIFICATION_ERR " +
                                    "The files will be deleted.")
                            throw MiniAppVerificationException(SIGNATURE_VERIFICATION_ERR)
                        }
                    }
                    
                    storage.saveFile(file, baseSavePath, apiClient.downloadFile(file).byteStream())
```

Reply:
Yeah, I'll update this part.